File: Missing And Repeating.java
class Solve {
    int[] findTwoElement(int arr[]) {
        // code here
         int n=arr.length;
        Arrays.sort(arr);
        int two[]=new int[n+1];
        int ans[]=new int[2];
        for(int i=0;i<n;i++){
        two[arr[i]]++;
        }
        for(int i=1;i<=n;i++){
        if(two[i]>1){
        ans[0]=i;
            }
        if(two[i]==0){
        ans[1]=i;
            }
        }
        return ans;
    }
}

File: Missing And Repeating.cpp
class Solution {
  public:
    vector<int> findTwoElement(vector<int>& arr) {
        // code here
        long n=arr.size();
        long aSum=(n*(n+1))/2;
        long aSum2=(n*(n+1)*(2*n+1))/6;
        long obSum=0;
        long obSum2=0;
        for(long val:arr){
            obSum+=val;
            obSum2+=(val*val);
            
        }
        long val1=aSum-obSum;
        long val2=aSum2-obSum2;
        val2=val2/val1;
        int missing=(val1+val2)/2;
        int repeating =val2-missing;
        return {repeating,missing};
    }
};

